tests/test_cvi.py

```diff
@@ -11,41 +11,206 @@
 import os
 from pathlib import Path
 import logging as lg
+# import inspect
 
 # --------------------------------------------------------------------------- #
 # CUSTOM IMPORTS
 # --------------------------------------------------------------------------- #
 
 import pytest
 import numpy as np
+import pandas as pd
 
 # --------------------------------------------------------------------------- #
 # LOCAL IMPORTS
 # --------------------------------------------------------------------------- #
+
 print(f"\nTesting path is: {os.getcwd()}")
 import src.cvi as cvi
 # import ..
 
+# --------------------------------------------------------------------------- #
+# FIXTURES
+# --------------------------------------------------------------------------- #
+
+# Set the fixture scope to the testing session to load the data once
+@pytest.fixture(scope="session")
+def data() -> dict:
+    """
+    Data loading test fixture.
+
+    This fixture is run once for the entire pytest session.
+    """
+
+    p = 0.1
+    lg.info("LOADING DATA")
+
+    data_path = Path("tests", "data")
+    # Load the test datasets
+    # correct = pd.read_csv(data_path.joinpath("correct_partition.csv")).sample(frac=p).convert_dtypes()
+    # over = pd.read_csv(data_path.joinpath("over_partition.csv")).sample(frac=p).convert_dtypes()
+    # under = pd.read_csv(data_path.joinpath("under_partition.csv")).sample(frac=p).convert_dtypes()
+    correct = pd.read_csv(data_path.joinpath("correct_partition.csv")).sample(frac=p).sort_index()
+    over = pd.read_csv(data_path.joinpath("over_partition.csv")).sample(frac=p).sort_index()
+    under = pd.read_csv(data_path.joinpath("under_partition.csv")).sample(frac=p).sort_index()
+
+    # Coerce the dataframe as two numpy arrays each for ease
+    correct_samples = correct.to_numpy(dtype=float)[:, 0:2]
+    correct_labels = correct.to_numpy(dtype=int)[:, 2] - 1
+    over_samples = over.to_numpy(dtype=float)[:, 0:2]
+    over_labels = over.to_numpy(dtype=int)[:, 2] - 1
+    under_samples = under.to_numpy(dtype=float)[:, 0:2]
+    under_labels = under.to_numpy(dtype=int)[:, 2] - 1
+
+    # Construct the dataset dictionary
+    data_dict = {
+        "datasets": {
+            "correct": {
+                "samples": correct_samples,
+                "labels": correct_labels,
+            },
+            "over": {
+                "samples": over_samples,
+                "labels": over_labels,
+            },
+            "under": {
+                "samples": under_samples,
+                "labels": under_labels,
+            },
+        },
+        "counts": {
+            "correct": len(correct.index),
+            "over": len(over.index),
+            "under": len(under.index),
+        },
+    }
+
+    return data_dict
+
+# --------------------------------------------------------------------------- #
+# UTILITY FUNCTIONS
+# --------------------------------------------------------------------------- #
+
+def get_cvis() -> list:
```

Comment:
use "from Typing import List" and the define the return type with -> List[cvi]

Reply:
While I like that idea, I was wondering: would this typing fail for inheriting classes? All my CVI implementations inherit from a base CVI type, but does that mean that a linter would get upset that they are not strictly speaking CVIs?